tests/unit_tests/scheduler/test_scheduler.py

```diff
@@ -333,6 +333,36 @@ async def kill():
     assert scheduler_task.result() == EVTYPE_ENSEMBLE_CANCELLED
 
 
+@pytest.mark.timeout(6)
+async def test_that_job_does_not_retry_when_killed(realization, mock_driver):
+    kill_me = asyncio.Event()
+
+    retries = 0
+
+    async def wait():
+        nonlocal retries
+        retries += 1
+        kill_me.set()
+        await asyncio.sleep(1000)
+
+    sch = scheduler.Scheduler(mock_driver(wait=wait), [realization], max_submit=2)
+
+    scheduler_task = asyncio.create_task(sch.execute())
+
+    await kill_me.wait()
+    await sch.cancel_all_jobs()
+
+    await scheduler_task
+    assert scheduler_task.result() == EVTYPE_ENSEMBLE_CANCELLED
+    assert retries == 1
```

Comment:
I suggest changing the line to `assert retries == 1, "Job` was resubmitted after killing"` - self-documenting :)